gui/drawer.go

```diff
@@ -125,17 +126,22 @@ L:
 	d.chartDrawing.Store(false)
 }
 
-func (d *drawer) redrawGauge(ctx context.Context, maxSize int) {
-	var count float64
-	size := float64(maxSize)
+func (d *drawer) redrawGauge(ctx context.Context, duration time.Duration) {
+	totalTime := float64(duration)
 	d.widgets.progressGauge.Percent(0)
-	for {
+	for start := time.Now(); ; {
 		select {
 		case <-ctx.Done():
 			return
 		case <-d.gaugeCh:
```

Comment:
Let's remove the `gaugeCh` field from drawer.

```suggestion
		case <-ticker.C:
```